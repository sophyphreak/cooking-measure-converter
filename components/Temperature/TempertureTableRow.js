import React from 'react'

import { Label } from 'reactstrap'
import AmountInput from '../TableRowComponent/AmountInput/AmountInput'

const TemperatureTableRow = ({
  metricLabel,
  metricInput,
  imperialInput,
  imperialLabel
}) => (
  <tr>
    <th>
      <Label className="align-right">{metricLabel}</Label>
    </th>
    <td>
      <AmountInput value={metricInput.value} onChange={metricInput.onChange} />
    </td>
    <td>
      <AmountInput
        value={imperialInput.value}
        onChange={imperialInput.onChange}
      />
    </td>
    <th>
      <Label>{imperialLabel}</Label>
    </th>
  </tr>
)

export default TemperatureTableRow
