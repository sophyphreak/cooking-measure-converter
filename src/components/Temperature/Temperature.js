import React from 'react';
import TemperatureTableRow from './TempertureTableRow';

const Temperature = ({ temperatureProps }) => {
  const {
    temperatureState,
    onImperialTemperatureChange,
    onMetricTemperatureChange
  } = temperatureProps;
  const { imperialTemperature, metricTemperature } = temperatureState;
  const metricLabel = '°c';
  const metricInput = {
    value: metricTemperature,
    onChange: onMetricTemperatureChange
  };
  const imperialInput = {
    value: imperialTemperature,
    onChange: onImperialTemperatureChange
  };
  const imperialLabel = '°f';
  return (
    <TemperatureTableRow
      metricLabel={metricLabel}
      metricInput={metricInput}
      imperialInput={imperialInput}
      imperialLabel={imperialLabel}
    />
  );
};

export default Temperature;
