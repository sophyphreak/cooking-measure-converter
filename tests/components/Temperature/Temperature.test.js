import React from 'react';
import { shallow } from 'enzyme';
import Temperature from '../../../src/components/Temperature/Temperature';

let temperature,
  onImperialTemperatureChange,
  onMetricTemperatureChange,
  temperatureProps,
  wrapper;

beforeEach(() => {
  temperature = {
    imperialTemperature: '',
    metricTemperature: ''
  };
  onImperialTemperatureChange = jest.fn();
  onMetricTemperatureChange = jest.fn();
  temperatureProps = {
    temperature,
    onImperialTemperatureChange,
    onMetricTemperatureChange
  };
  wrapper = shallow(<Temperature temperatureProps={temperatureProps} />);
});

test('should render Temperature correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
