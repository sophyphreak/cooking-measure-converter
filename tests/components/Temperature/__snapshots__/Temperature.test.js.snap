// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Temperature correctly 1`] = `
<TableRow
  imperialUnit="F"
  imperialValue=""
  metricUnit="C"
  metricValue=""
  onImperialChange={[MockFunction]}
  onMetricChange={[MockFunction]}
/>
`;
